crates/solver/src/settlement_access_list.rs

```diff
@@ -151,7 +213,63 @@ impl AccessListEstimating for TenderlyApi {
                 .map(Into::into)
                 .collect())
         }))
-        .await
+        .await)
+    }
+}
+
+fn resolve_call_request(tx: &TransactionBuilder<DynTransport>) -> Result<(H160, H160, Bytes)> {
+    let from = tx
+        .from
+        .clone()
+        .context("transaction from does not exist")?
+        .address();
+    let to = tx.to.context("transaction to does not exist")?;
+    let data = tx.data.clone().context("transaction data does not exist")?;
+    Ok((from, to, data))
+}
+
+// this function should remove duplicates and elements that are not useful
+// currently only eliminating addresses and storages with value '1'
+// should be updated continually as we learn more about the imperfections of 3rd party access_list calculators
+#[allow(dead_code)]
+fn filter_access_list(access_list: AccessList) -> AccessList {
```

Comment:
This functionality seems specific to each `AccessListEstimating` implementation using a 3rd party service. This is probably fine for now but in case we find more imperfections unique to each implementation it would make sense to make this a member function of `NodeApi` and `TenderlyApi`.

Reply:
You are right. :)

Reply:
Can we add a unit test for this filtering function (might help me understand how it was supposed to work)? I'm not sure I understand exactly what its why `address(1)` and `bytes32(1)` slots are special.

Also, if this is specific to a node/tenderly, can we prefix the method name with `erigon_...` or `tenderly_...` for example?

Reply:
We have two tests. I renamed them now to be more clear, `filter_access_list_node` and `filter_access_list_tenderly`.
The `filter_access_list` is used for both apis.

Reply:
And also I am not sure why `1` is returned, would probably need to dig into evm

Reply:
I would like to have an explanatory comment for the 1 address and 1 hash too.  It should explain why you included this filter. If the answer is that this some quirk of the nodes then that's fine but should still be a comment.

Reply:
> And also I am not sure why 1 is returned

Note that `address(1)` is the precompile that we use for signature recovery in our settlement contract. Likely related.

Reply:
Added the comment.